lib/phoenix_live_view/utils.ex

```diff
@@ -282,6 +282,8 @@ defmodule Phoenix.LiveView.Utils do
   Calls the `c:Phoenix.LiveView.mount/3` callback, otherwise returns the socket as is.
   """
   def maybe_call_live_view_mount!(%Socket{} = socket, view, params, session) do
+    {:ok, %Socket{} = socket} = Lifecycle.mount(params, session, socket)
```

Comment:
is this handling the case where a mount hook halts/redirects? It looks like we allow it to continue here until handle_params

Reply:
It is now :) lmk if it's what you were expecting – basically we raise if you try to `{:cont socket}` with a redirect or `{:halt, socket}` without one.